Public/Src/Cache/DistributedCache.Host/Service/IDistributedCacheServiceHost.cs

```diff
@@ -36,6 +37,6 @@ public interface IDistributedCacheServiceHost
         /// <summary>
         /// Retrieves secrets from key vault
         /// </summary>
-        Task<Dictionary<string, string>> RetrieveKeyVaultSecretsAsync(List<string> secrets, CancellationToken token);
+        Task<Dictionary<string, Credentials>> RetrieveKeyVaultSecretsAsync(List<RetrieveSecretsRequest> requests, CancellationToken token);

```

Comment:
Let's discuss this offline, but I do believe that the result should be a secret not credentials.

And I still believe that it make sense to make the interface between bxl and CB as small as possible and it means that I would prefer to deal with type 'SecretKind' = PlaintText | Sas; and the result could be just a string actually. And it is a client's responsibility to parse the string appropriately.

Reply:
I can do this, but the trade-off is we will need to expose an updating token type and will need to implement glue with IUpdatingPlainText/etc in the CloudBuild side.

---
In reply to: [303562855](https://github.com/microsoft/BuildXL/pull/587#discussion_r303562855) [](ancestors = 303562855)